input/otlp/metrics.go

```diff
@@ -50,28 +51,45 @@ import (
 
 // ConsumeMetrics consumes OpenTelemetry metrics data, converting into
 // the Elastic APM metrics model and sending to the reporter.
-func (c *Consumer) ConsumeMetrics(ctx context.Context, metrics pmetric.Metrics) error {
+// The returned Result contains the number of accepted and rejected data points.
+// A non-nil error indicates that at least 1 data point is rejected.
+func (c *Consumer) ConsumeMetrics(ctx context.Context, metrics pmetric.Metrics) (Result, error) {
+	totalCount := int64(metrics.DataPointCount())
 	if err := c.sem.Acquire(ctx, 1); err != nil {
-		return err
+		return parseResultFromError(err, totalCount)
 	}
 	defer c.sem.Release(1)
 
 	receiveTimestamp := time.Now()
 	c.config.Logger.Debug("consuming metrics", zap.Stringer("metrics", metricsStringer(metrics)))
-	batch := c.convertMetrics(metrics, receiveTimestamp)
-	return c.config.Processor.ProcessBatch(ctx, batch)
+	var err error
+	batch, droppedDataPoints := c.convertMetrics(metrics, receiveTimestamp)
+	if droppedDataPoints > 0 {
+		err = errors.New("unsupported data points")
+	}
+	processBatchErr := c.config.Processor.ProcessBatch(ctx, batch)
+	if processBatchErr == nil {
+		return Result{Rejected: droppedDataPoints, Accepted: totalCount - droppedDataPoints}, err
+	}
+	if errorWithResult, ok := processBatchErr.(ErrorWithResult); ok {
+		return Result{
+			Rejected: droppedDataPoints + errorWithResult.Rejected,
+			Accepted: totalCount - (droppedDataPoints + errorWithResult.Rejected),
+		}, errors.Join(err, processBatchErr)
+	}
+	return Result{Rejected: totalCount}, processBatchErr
```

Comment:
[For reviewer] This is a little messier than I like, but this is needed in order to account for rejected counts at each level. Could be a little easier if we don't track Accepted.